### PR TITLE
New device agnostic option to avoid the suffix in the snapshots

### PR DESCRIPTION
**Pull request by augustorsouza**
Adding an option to the agnostic snapshots to avoid the addition of the suffixes regarding dimension.

### CODE REVIEW

**Comment by augustorsouza** [comment]
Is there anyone looking at pull requests for this project?

**Comment by alanzeino** [comment]
I didn't realise this when I commented on the issue, but this is too confusing because `FBSnapshotTestCaseAgnosticOptionNone` already exists. 

And there's no nice way to do this with the option mask, because if we add a _new_ `FBSnapshotTestCaseAgnosticOptionScale` then everyone will update to the new version and wonder why the scale isn't being added.

Perhaps it might work if we bump the major library version to follow semantic versioning, but that still doesn't solve the problem that people will wonder why all their tests are failing once they update.

**Comment by augustorsouza** [comment]
@alanzeino I don't know if I understood everything you commented. But, if people update right now with the fix I did they will have to add an option called `FBSnapshotTestCaseAgnosticOptionNoSuffix` to avoid the automatic append of the suffix.

If they still avoid using any option the default behavior will be used.

**Comment by alanzeino** [comment]
Sorry, should've been clearer. The confusing part is that there will be two options; `FBSnapshotTestCaseAgnosticOptionNone` and `FBSnapshotTestCaseAgnosticOptionNoSuffix`. The right way would be to call the new one `FBSnapshotTestCaseAgnosticOptionScreenScale`, so that users opt in to being 'agnostic' on screen scale, but then anyone who updates to the version that adds that will have to add this option, or their existing tests will fail.

I believe that in general the reason why the original library never made device scale something you can remove, is because the original authors would have considered that to be a fundamental part of the file name.

In short looking at this more now, I don't think we can make the change in a way that works for existing code.

**Comment by acecilia** [comment]
I would like to have this. Personally, I am manually adding the device name to the identifier, so the 2x/3x suffix is useless. 
Seeing that sometimes using two different simulators with the same scaling throw different results, the scaling indicator is not as useful as it was when the library was conceived.

In addition, the actual implementation does not allow to modify the order of the agnostic suffixes/prefixes, or which separators to use (underscore, dots, brackets..). I would rewrite the implementation, so instead of having different flags for the different suffixes/prefixes, we can pass a formatter, which allows to modify the order and the separators (similar to the date formatters).

**Comment by augustorsouza** [comment]
sounds good, what do you think @alanzeino ? I can change my implementation to some kind of formatter block that optionally anyone can send to format the file name of the screenshot in any scenario the developer using the lib could chose.

**Comment by alanzeino** [comment]
_(Just investigating device names a bit more, seems like you can't get a more precise name when using the simulator. What I suggested above might not work. Will keep looking.)_

**Comment by acecilia** [comment]
@alanzeino the main problem for me is that the framework does not allow for a flexible customization of the file names (I dont like the separators and order that the framework is using), so what I did is pass my own identifier, created using DeviceKit:

```
let dev = Device()
let imageSize = "\(Int(imageView.frame.size.width))x\(Int(imageView.frame.size.height))"
let identifier = "[\(dev.deviceName)]-[\(dev.systemVersion)]-[\(imageSize)]-\(snapshotIndex)-\(title)"
```

 It looks like this:  

`<testIdentifierIntroducedByTheFramework>_[iPhone 8]-[11.4]-[375x667]-<snapshotIndexForCurrentTest>-<Description of the screenshot>@2x.png`

And a real example:

`testSignup_[iPhone 8]-[11.4]-[375x667]-1-phone prefix selector@2x.png`

As you can see, the things I could not get rid off is the _test name_ and the _resolution suffix_. My ideal snapshot name would be:

`Signup-[iPhone 8]-[11.4]-[375x667]-1-phone prefix selector.png`

IMHO what makes sense to me is one of this options (in order of preference):
- Keep the current naming convention and implementation, but adding the ability to override it 100% [allow removing the test name and the resolution indicator] in a easier way [right now getting rid of all naming modifications introduced by the framework requires to set configuration here and there, instead of one only "disableAllIdentifierCustomizations"].
- Do not provide any kind of naming convention, and require the user to pass a custom identifier, which will be used as is, and without any modifications (nor agnostic or other). Example: if the identifier is "image1", the image should be called "image1.png", and not "testName_image1@2x.png". In this case the user is on charge of properly naming the images.
- Change the implementation to some kind of formatter that allows flexibility and easy customization of the identifier. Even in this option, I think that people that want a very specific image identifier will probably require to pass their own. Also, implementing a formatter that provides with such flexibility + being type safe, may be too much effort for this tiny problem.

**Comment by acecilia** [comment]
Implementing 100% custom name files by allowing to remove the _test name_ and the _resolution indicator_ will not break any API, it will be an incremental change, so you dont need to deprecate anything. It will not require to change main version number.

**Comment by alanzeino** [comment]
Unfortunately long term we will need to deprecate something because this API is already pretty bloated with both `deviceAgnostic` and `agnosticOptions`, and will become more so if we add a custom formatter option.

**Comment by augusto-souza-ifood** [comment]
Another option for us to think: optional env vars in the scheme, like the setting for the directories of the reference images.

Something like: `USE_IMAGE_SUFFIX` that accepts `true` or `false` The default is what people use now, so if it is not setted and we assume `true`, but we can turn it off setting this to `false`.

**Comment by augustorsouza** [comment]
@alanzeino and @acecilia if you guys agree I will can implement a environemnt var based solution. :)

**Comment by alanzeino** [comment]
An update here. In #64 we're going to add a new option mask to replace `agnosticOptions` called `fileNameOptions`. I'm going to add a new option for the screen scale, to take advantage of the opportunity presented by adding this new mask.
